### DISCUSSION TITLE
Memory leak SaaS

### DISCUSSION THREAD

**Discussion by GDTNguyen** [vercel/next.js#54803] [Help]
### Summary

Introduction

I built a SaaS website and I'm currently having issues with memory, specifically my baseline memory usage is 500mb and when users start using the application it reaches 1.5 - 2 GB memory usage. 

Specifications

Next.js:	v.13.2.3
React:	v.18.2
Node local:	v.18.12.1

![Screenshot 2023-08-30 at 23 28 38](https://github.com/vercel/next.js/assets/16839142/dba95211-ae9a-44d2-8882-e35d75c9f1ef)

I've already looked at node_modules and identified packages that are large(100mb ect...), checked globals, and properly handle database connections(prisma). 

How do I identify and debug memory usage on a service like heroku especially for a frontend framework like next js?

**Comment by ivanhueso** [comment]
Having similar issue, since im oved from pages to app Next.js 13.5.x

**Comment by kdelemme** [comment]
I'm having a similar issue... I'm currently thinking the issue is around the image caching with sharp.

**Comment by TheodoreRed** [comment]
"I built a SaaS website and I'm currently having issues with memory, specifically my baseline memory usage is 500mb and when users start using the application it reaches 1.5 - 2 GB memory usage."

Yeah I am seeing the exact numbers on my Next.js App Router v15.2.3 app
